jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -817,6 +817,54 @@ ecma_builtin_helper_def_prop_by_index (ecma_object_t *obj_p, /**< object */
   return ret_value;
 } /* ecma_builtin_helper_def_prop_by_index */
 
+/**
+ * Helper function for at() functions.
+ *
+ * See also:
+ *          ECMA-262 Stage 3 Draft Relative Indexing Method proposal 3. 4. 5. 6.
+ *
+ * Used by:
+ *         - The Array.prototype.at routine.
+ *         - The String.prototype.at routine.
+ *         - The TypedArray.prototype.at routine.
+ *
+ * @return ecma number
+ *
+ */
+ecma_number_t
+ecma_builtin_helper_check_index (ecma_value_t index, /**< index argument */
```

Comment:
this method must return with ecma_value_t since it may raises error what need to be checked at the caller site..

How about this one?
```
ecma_value_t
ecma_builtin_helper_check_index (ecma_value_t index,
                                 ecma_number_t* len)
```